GMX-Bots/Large-Open-and-Increase-Positions/src/agent.spec.ts

```diff
@@ -0,0 +1,549 @@
+import { Finding, ethers, FindingSeverity, FindingType, HandleTransaction, TransactionEvent } from "forta-agent";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools/lib/tests";
+import { provideTransactionHandler } from "./agent";
+import NetworkData from "./network";
+import { INCREASE_POSITION_EVENT, UPDATE_POSITION_EVENT } from "./constants";
+import { ethersBnToBn } from "./utils";
+
+const TEST_PRICE_PRECISION = 30;
+const MOCK_OTHER_EVENT: string = "event UpdateFundingRate(address token, uint256 fundingRate)";
+const MOCK_EVENT_ABI: string[] = [INCREASE_POSITION_EVENT, UPDATE_POSITION_EVENT];
+const MOCK_TOKEN: string = createAddress("0x85e");
+const MOCK_ACCOUNT: string = createAddress("0x11a");
+const MOCK_OTHER_CONTRACT: string = createAddress("0x99");
+const MOCK_VAULT_ADDRESS: string = createAddress("0xa1");
+const MOCK_IFACE: ethers.utils.Interface = new ethers.utils.Interface(MOCK_EVENT_ABI);
+const key: string = ethers.utils.formatBytes32String("positionKey");
+
+const smallSize = ethers.BigNumber.from(1000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION));
+const size1 = ethers.BigNumber.from(500000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+const size2 = ethers.BigNumber.from(600000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+const size3 = ethers.BigNumber.from(700000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+const size4 = ethers.BigNumber.from(8000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+const size5 = ethers.BigNumber.from(90000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+const size6 = ethers.BigNumber.from(90000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+
+const sizeDelta1 = ethers.BigNumber.from(50000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+const sizeDelta2 = ethers.BigNumber.from(60000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+const sizeDelta3 = ethers.BigNumber.from(70000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+const sizeDelta4 = ethers.BigNumber.from(8000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+const sizeDelta5 = ethers.BigNumber.from(90000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
+const sizeDelta6 = ethers.BigNumber.from(90000000).mul(ethers.BigNumber.from(10).pow(TEST_PRICE_PRECISION - 3));
```

Comment:
The current version is fine. But for the future consider thinking about merging test data in a single array, in a format that is easier to read. This will faciliate understanding what each test case if referring to (large opening/ large increase .